src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketPostClient.java

```diff
@@ -0,0 +1,21 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import okhttp3.Response;
+
+/**
+ * Basic Bitbucket post client. A client is normally threadsafe and can be used multiple times.
+ *
+ * @param <T> The model used in the post parameters
+ */
+public interface BitbucketPostClient<T> {
```

Comment:
I used a generic here to represent the model provided to the POST call, but I think given Jackson doesn't require a particular type to map objects to JSON, this is unnecessary.